### PR TITLE
Apply worker pool to speed up plan-preview's execution

### LINKED ISSUE #2296: Reduce plan-preview execution time by making a pool to build result concurrently
**What would you like to be added**:

For those who use Kustomize with remote sources, `kustomize build` command will take a long time to run.
 
https://github.com/pipe-cd/pipe/blob/master/pkg/app/piped/planpreview/builder.go#L162

**Why is this needed**:

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2296

**Does this PR introduce a user-facing change?**:

```release-note
Apply worker pool to speed up plan-preview's execution
```

### CODE REVIEW

File: pkg/app/piped/planpreview/builder.go [hunk 1]
```diff
@@ -158,82 +160,121 @@ func (b *builder) build(ctx context.Context, id string, cmd model.Command_BuildP
...
+	}
 
-		// We only need the environment name
-		// so the returned error can be ignorable.
-		var envName string
-		if env, err := b.environmentGetter.Get(ctx, app.EnvId); err == nil {
-			envName = env.Name
-		}
+	// Start some workers to speed up building time.
+	b.logger.Info(fmt.Sprintf("start %d workers for building plan-preview results for %d applications", numWorkers, numApps))
```
Comment (khanhtc1202) [inline pkg/app/piped/planpreview/builder.go:179]: should we use the above wrapped `logger`? 👀

Comment (nghialv) [reply pkg/app/piped/planpreview/builder.go:179]: Right. Nice catch!

File: pkg/app/piped/planpreview/builder.go [hunk 2]
```diff
@@ -158,82 +160,121 @@ func (b *builder) build(ctx context.Context, id string, cmd model.Command_BuildP
...
-	for _, app := range triggerApps {
-		b.logger.Info("will decide sync strategy for an application",
-			zap.String("id", app.Id),
-			zap.String("name", app.Name),
-			zap.String("kind", app.Kind.String()),
-		)
+	var (
+		numApps  = len(triggerApps)
+		appCh    = make(chan *model.Application, numApps)
+		resultCh = make(chan *model.ApplicationPlanPreviewResult, numApps)
```
Comment (khanhtc1202) [inline pkg/app/piped/planpreview/builder.go:171]:
```suggestion
		appCh    = make(chan<- *model.Application, numApps)
		resultCh = make(<-chan *model.ApplicationPlanPreviewResult, numApps)
```
nit

Comment (khanhtc1202) [reply pkg/app/piped/planpreview/builder.go:171]: My missed 👌 never mind

**Comment by khanhtc1202** [comment]
Nice work
/lgtm

**Comment by nakabonne** [comment]
Good improvement
/approve

**Pull request merged**